python-ethtool/ethtool.c

```diff
@@ -755,7 +757,6 @@ static PyObject *__struct_desc_create_dict(struct struct_desc *table,
     return dict;
 free_dict:
     goto out;
-    dict = NULL;
```

Comment:
The fix does not look correct -- why would you have two distinct labels with the same semantics?

Should not you actually free `dict`?
```
out:
    return dict;                                                                              
free_dict:
    Py_DECREF(dict);                                                                          
    return NULL;                                                                              
}
```

If not, you can just write `goto out;` instead of `goto free_dict;` to make it clear what the code does.

Reply:
Yes, you are right. I am neither creator nor an expert of this code so I just wanted to remove unreachable code. But after your review, it makes sense to read it again and think about it more.

I fixed it and I removed both labels to make it clear. What do you think about it now?

Reply:
Looks better now.  Thanks for the update!